### PR TITLE
Alert History Empty State Default Card Size / Shows With Empty Nodes

### PR DESCRIPTION
**Pull request by jennifer-notifi**
- If nodes does not exist OR length is zero, show empty card state
- Subscription Card default height size is 450px, width is 300px
- Fix - If no telegram or text message, do not show an extra separator

Configuration with no alert history (and two out of three inputs):

Screen Shot 2022-11-16 at 10 49 57 AM

Screen Shot 2022-11-16 at 10 50 52 AM

Screen Shot 2022-11-16 at 10 51 07 AM

Configuration with alert history (and all inputs):

Screen Shot 2022-11-16 at 10 54 43 AM

Screen Shot 2022-11-16 at 10 55 33 AM

Screen Shot 2022-11-16 at 2 25 51 PM

Alert History Icons and Date Padding preserved even on narrow screens:
Screen Shot 2022-11-18 at 9 43 12 AM

### CODE REVIEW

File: packages/notifi-react-card/lib/components/subscription/defaults.css [hunk 1]
```diff
@@ -149,6 +149,8 @@
   line-height: 14px;
   color: var(--notifi-secondary-font-color);
   text-align: center;
+  height: 400px;
+  line-height: 400px;
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/components/subscription/defaults.css:153]: I think we shouldn't use line-height to try and center the text.

Also, the alert description shouldn't take up a height of 400px, instead, the base card should have a default height defined and this component should grow to fill the  space.

File: packages/notifi-react-card/lib/components/subscription/subscription-card-views/HistoryCardView.tsx [hunk 2]
```diff
@@ -148,7 +148,7 @@ export const AlertHistoryView: React.FC<AlertHistoryViewProps> = ({
           classNames?.dividerLine,
         )}
       />
-      {alertHistoryData?.nodes ? (
+      {alertHistoryData?.nodes && alertHistoryData?.nodes?.length > 0 ? (
```
Comment (juni-notifi) [inline packages/notifi-react-card/lib/components/subscription/subscription-card-views/HistoryCardView.tsx:151]: i'd prefer to see this pulled out so we don't have an implicit conversion

e.g.: on line 128, add `const nodeLength = alertHIstoryData?.nodes?.length ?? 0;` and here, use `nodeLength > 0`

**Review by juni-notifi — CHANGES_REQUESTED** [review]
Would you mind making the card have a fixed height and using flex-grow to size the contents instead?

File: packages/notifi-react-card/lib/components/subscription/subscription-card-views/HistoryCardView.tsx (hunk 2 above)
Comment (jennifer-notifi) [reply packages/notifi-react-card/lib/components/subscription/subscription-card-views/HistoryCardView.tsx:151]: Thank you for letting me know about implicit conversions.

I learned that implicit conversion happens when expression contains operands of different built-in types, and no explicit casts are present, the compiler converts one of the operands so that the types match. 

The reason why `const nodeLength = alertHIstoryData?.nodes?.length ?? 0`; will not run into implicit conversion issues is because it will return 0 if `alertHistoryData?.nodes?.length` is `null `or `undefined `or `alertHistoryData?.nodes?.length value` if it is not null or undefined. 

In other words, it will always return a number.

**Review by Nathan-Notifi — APPROVED** [review]

**Comment by jennifer-notifi** [comment]
Apologies, due to the prioritization of Acala wallets, I forgot about this PR. I will review and make the updates and/or open a different PR by EOD Monday.

**Comment by Nathan-Notifi** [comment]
good to know@ looks like there are some conflicts so i'd rebase as well.

**Comment by jennifer-notifi** [comment]
Opening a new PR
